cli/src/commands/git/clone.rs

```diff
@@ -76,6 +76,13 @@ pub struct GitCloneArgs {
     /// while all subsequent fetches will only fetch included tags.
     #[arg(long, value_enum)]
     fetch_tags: Option<FetchTagsMode>,
+    /// Name of the branch to fetch and use as the parent of the working-copy
+    /// change
+    ///
+    /// If not present, all branches are fetched and the repository's default
+    /// branch is the parent of the working-copy change
+    #[arg(long)]
+    branch: Option<String>,
```

Comment:
Maybe we can support multiple branches, i.e. `branch: Vec<StringPattern>`?

If https://github.com/jj-vcs/jj/pull/7275 gets merged, these patterns can be set as the default refspecs so future `jj git fetch` wouldn't fetch all branches.

I'm not sure which branch should be checked out if there are multiple matching bookmarks, but maybe we can try the default first, then fall back to one of the matching bookmark?

Reply:
Multiple branches makes sense.

I'm thinking checking out the first matching pattern could be nice. It gives the caller control over the branch which I feel is the purpose of this option.

So `jj git clone <repo> --branch feature_branch --branch main --branch other` would result in `feature_branch` being checked out.

Reply:
That sounds also good, but well need to pick one arbitrarily for `--branch glob:pattern`. I don't mind if the first matching branch (in e.g. lexicographical order) were chosen if that's easier to implement.

Reply:
Added multiple branch support, where the first match in lexicographical order, of the first `--branch` pattern by position, becomes the working copy parent.